core/api/views/business_plan.py

```diff
@@ -49,25 +53,45 @@ class BusinessPlanViewSet(
     viewsets.GenericViewSet,
 ):
     permission_classes = [IsUserAllowedBP]
-    serializer_class = BusinessPlanSerializer
     filterset_class = BusinessPlanFilter
     filter_backends = [
         DjangoFilterBackend,
         filters.OrderingFilter,
         filters.SearchFilter,
     ]
+    ordering = ["agency__name", "id"]
     ordering_fields = "__all__"
     search_fields = ["agency__name"]
 
     def get_queryset(self):
         business_plans = BusinessPlan.objects.all()
 
+        if self.action == "get":
+            self.filterset_class = BPRecordFilter
```

Comment:
hmmm get_filterset_class did not worked? It is a little confusing that the filterset class is set in get_queryset method